checks.go

```diff
@@ -0,0 +1,66 @@
+package mackerel
+
+// CheckStatuses
+const (
+	CheckStatusOK       = "OK"
+	CheckStatusWarning  = "WARNING"
+	CheckStatusCritical = "CRITICAL"
+	CheckStatusUnknown  = "UNKNOWN"
+)
+
+// CheckReport represents a report of check monitoring
+type CheckReport struct {
+	Source               CheckSource `json:"source"`
+	Name                 string      `json:"name"`
+	Status               string      `json:"status"`
+	Message              string      `json:"message"`
+	OccurredAt           int64       `json:"occurredAt"`
+	NotificationInterval uint        `json:"notificationInterval,omitempty"`
+	MaxCheckAttempts     uint        `json:"maxCheckAttempts,omitempty"`
+}
+
+// CheckSource represents interface to which each check source type must confirm to
+type CheckSource interface {
+	CheckType() string
+
+	isCheck()
+}
+
+const checkTypeHost = "host"
+
+// Ensure each check type conforms to the CheckSource interface.
+var _ CheckSource = (*checkSourceHost)(nil)
+
+// Ensure only checkSource types defined in this package can be assigned to the
+// CheckSource interface.
+func (cs *checkSourceHost) isCheck() {}
+
+type checkSourceHost struct {
+	Type   string `json:"type"`
+	HostID string `json:"hostId"`
+}
+
+// CheckType is for satisfying CheckSource interface
+func (cs *checkSourceHost) CheckType() string {
+	return checkTypeHost
+}
+
+// NewCheckSourceHost returns new CheckSource which check type is "host"
+func NewCheckSourceHost(hostID string) CheckSource {
+	return &checkSourceHost{
+		Type:   checkTypeHost,
+		HostID: hostID,
+	}
+}
+
+// CheckReports represents check reports for API
+type CheckReports struct {
+	Reports []*CheckReport `json:"reports"`
+}
+
+// ReportCheckMonitors reports check monitoring results
+func (c *Client) ReportCheckMonitors(crs *CheckReports) error {
```

Comment:
I'm not sure the method name is appropriate... Maybe `PostCheckReports` is better?